core/src/BibleEngine.class.ts

```diff
@@ -575,6 +588,48 @@ export class BibleEngine {
         return { ...book, ...updates };
     }
 
+    async downloadBooks(books: IBibleBookEntity[], fileUrl: string, version: BibleVersionEntity) {
```

Comment:
i see a general issue here and one specific to this method. concerning the latter i'd say this method does not what it says and it does too many things. i guess from the user or UI level of things "download" includes actually importing the data into the database but i'd argue that from a code level point of view i'd rather name the method `downloadAndImportBooks`. And then again if you already want to have an all-purpose method, i wonder why you would need to pass `books`, since what this method actually does is downloading and importing a whole version: `downloadAndImportVersion(uid: string)`

This brings me to the more general point: similar to what i said above i would recommend putting this functionality on your app-level bible service. The way you download stuff (and even cache data, if at all) is very app specific. If you really need it in the library, i think it would rather belong to `BibleEngineClient`, since we migrated all remote-access and other client-related stuff to there.

Reply:
removed this logic, I'll put up a followup PR for this